src/Kernel32/Kernel32.cs

```diff
@@ -375,6 +376,35 @@ public static partial class Kernel32
             [Friendly(FriendlyFlags.Out | FriendlyFlags.Optional)] int* lpNumberOfBytesWritten,
             OVERLAPPED* lpOverlapped);
 
+        /// <summary>
+        /// Moves the file pointer of the specified file.
+        /// </summary>
+        /// <param name="hFile">
+        /// A handle to the file.
+        /// The file handle must be created with the GENERIC_READ or GENERIC_WRITE access right.
+        /// </param>
+        /// <param name="liDistanceToMove">
+        /// The number of bytes to move the file pointer. A positive value moves the pointer forward in the file and a negative value moves the file pointer backward.
+        /// </param>
+        /// <param name="lpNewFilePointer">
+        /// A pointer to a variable to receive the new file pointer. If this parameter is <see langword="null"/>, the new file pointer is not returned.
+        /// </param>
+        /// <param name="dwMoveMethod">
+        /// The starting point for the file pointer move.
+        /// </param>
+        /// <returns>
+        /// If the function succeeds, the return value is nonzero.
+        /// If the function fails, the return value is zero. To get extended error information, call GetLastError.
+        /// </returns>
+        /// <seealso href="https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-setfilepointerex"/>
+        [DllImport(api_ms_win_core_file_l1_2_0, SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool SetFilePointer(
+            SafeObjectHandle hFile,
+            long liDistanceToMove,
+            out long lpNewFilePointer,
```

Comment:
`LONG` in native code is still just a 32-bit integer. When native code _really_ means 64-bit they use `LONGLONG`.  So I suspect this fails until you change it to `int`.

Reply:
Thanks for catching this, the P/Invoke declaration was for `SetFilePointerEx` instead of `SetFilePointer`